front/app/api/users/useChangePassword.ts

```diff
@@ -0,0 +1,22 @@
+import { useMutation } from '@tanstack/react-query';
+import { CLErrorsJSON } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import { IUser, IChangePassword } from './types';
+
+export const changePassword = async ({
+  userId,
+  ...requestBody
+}: IChangePassword) =>
+  fetcher<IUser>({
+    path: `/users/${userId}`,
+    action: 'post',
+    body: { user: { ...requestBody } },
+  });
+
+const useChangePassword = () => {
+  return useMutation<IUser, CLErrorsJSON, IChangePassword>({
+    mutationFn: changePassword,
+  });
```

Comment:
Is there any validation that needs to happen? E.g. if they log out and then log back in again - is there an issue if the cache hasn't been invalidated?

Reply:
Good question! This is changing the password and I've transferred all existing invalidations from the original service to this one. So a tentative no on whether this is needed but let's pay attention to it in the functional test, as well.